service/history/queues/executable.go

```diff
@@ -234,6 +235,10 @@ func (e *executableImpl) HandleErr(err error) (retErr error) {
 func (e *executableImpl) IsRetryableError(err error) bool {
 	// this determines if the executable should be retried within one submission to scheduler
 
+	if err == shard.ErrShardClosed || shard.IsShardOwnershipLostError(err) {
+		return false
+	}
```

Comment:
unify these 2 errors?

Reply:
Moved the check for `ErrShardClosed` in to `shard.IsShardOwnershipLostError`